diffpy/pdfmorph/pdfmorph_api.py

```diff
@@ -0,0 +1,241 @@
+    if (rv_cfg.get('smear') is not None
+        and rv_cfg.get('baselineslope') is None):
+        rv_cfg['baselineslope'] = -0.5
+    # config dict defines initial guess of parameters
+    chain = morphs.MorphChain(rv_cfg)
+    # rgrid
+    chain.append(morphs.MorphRGrid())
+    # configure morph chain
+    for k in active_morphs:
+        morph_cls = morph_step_dict[k]
+        if k == 'smear':
+            [chain.append(el()) for el in morph_cls]
+            refpars.append('baselineslope')
+        else:
+            chain.append(morph_cls())
+        refpars.append(k)
+    # exclude fixed options
+    if fixed_operations:
+        if not isinstance(fixed_operations, Iterable):
+            fixed_operations = [fixed_operations]
+        for opt in fixed_operations:
+            refpars.remove(opt)
+    # define refiner
+    refiner = ref.Refiner(chain, xobj, yobj, xref, yref)
+    if pearson:
+        refiner.residual = refiner._pearson
+    if addpearson:
+       refiner.residual = refiner._addpearson
+    # execute morphing
+    if refpars and refine:
+        # This works better when we adjust scale and smear first.
+        if "smear" in refpars:
+            rptemp = ["smear"]
+            if "scale" in refpars:
+                rptemp.append("scale")
+            refiner.refine(*rptemp)
+        # Refine all params
+        refiner.refine(*refpars)
+    else:
+        # no operation if refine=False or refpars is empty list
+        chain(xobj, yobj, xref, yref)
+
+    # summary
+    rw = tools.getRw(chain)
+    pcc = tools.getPearson(chain)
+    # restore rgrid
+    chain[0] = morphs.Morph()
+    chain(xobj, yobj, xref, yref)
+    # print output
+    if verbose:
+        print("== INFO: Following steps are fixed during morphing ==:\n{}\n"
+              .format('\n'.join(fixed_operations)))
+        print("== INFO: Refined morph parameters ==:\n")
+        output = "\n".join(["# {} = {:.6f}".format(k, v) for k, v in \
+                rv_cfg.items() if v is not None])
+        output += "\n# Rw = {:.6f}".format(rw)
+        output += "\n# Pearson = {:.6f}".format(pcc)
+        print(output)
+
+    rv_dict = dict(morph_chain=chain, morphed_config=rv_cfg,
+                   rw=rw, pcc=pcc)
+    return rv_dict
+
+
+def plot_morph(chain, ax=None, **kwargs):
+    """function to plot pdfmorph result from a morph chain
+
+    Parameters
+    ----------
+    chain: diffpy.pdfmorph.morphs.morphchain.MorphChain
+        An instance of processed morph chain.
+    ax: matplotlib.axes.Axes, optinal
+        An instance of Axes class to plot morphing result.
+        Default to the axes return by ``matplotlib.pyplot.gca()``
+    kwargs:
+        Additional keyword arguements will be passed 
+        to ``ax.plot(...**kwargs)``
+
+    Returns
+    -------
+    l_list: list
+        A list of ``matplotlib.lines.Line2D`` objects representing 
+        the plotted data.
+    """
+    if not ax:
```

Comment:
in this case it makes no difference, but in general it is better to use
`if ax is None` to test for missing arguments.